pkg/gc/collector.go

```diff
@@ -0,0 +1,83 @@
+package gc
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+
+	"github.com/kumahq/kuma/pkg/core"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+)
+
+type collector struct {
+	rm         manager.ResourceManager
+	cleanupAge time.Duration
+	polling    time.Duration
+}
+
+func NewCollector(rm manager.ResourceManager, polling, cleanupAge time.Duration) component.Component {
+	return &collector{
+		cleanupAge: cleanupAge,
+		rm:         rm,
+		polling:    polling,
+	}
+}
+
+func (d *collector) Start(stop <-chan struct{}) error {
+	ticker := time.NewTicker(d.polling)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			if err := d.cleanup(); err != nil {
+				gcLog.Error(err, "unable to cleanup")
+				continue
+			}
+		case <-stop:
+			return d.cleanup()
```

Comment:
What is the reason for cleaning up on stop? I think it's better to just return from here